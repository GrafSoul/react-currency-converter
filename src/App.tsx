import React from 'react';
import { Routes, Route } from 'react-router-dom';

import { Provider } from 'react-redux';
import { store } from './store';

import Footer from './components/Footer';
import Header from './components/Header';
import NotFound from './components/NotFound';

import Converter from './Converter';
import Currencies from './Currencies';

import classes from './App.module.scss';

const App: React.FC = () => {
    return (
        <Provider store={store}>
            <div className={classes.container}>
                <Header />

                <main className={classes.content}>
                    <Routes>
                        <Route path="/" element={<Currencies />} />
                        <Route path="/converter" element={<Converter />} />
                        <Route path="*" element={<NotFound />} />
                    </Routes>
                </main>

                <Footer />
            </div>
        </Provider>
    );
};

export default App;
