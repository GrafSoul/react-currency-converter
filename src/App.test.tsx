/* eslint-disable testing-library/render-result-naming-convention */
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

describe('App component', () => {
    it('App render', () => {
        render(
            <MemoryRouter>
                <App />
            </MemoryRouter>,
        );
        expect(screen.getByText('Выберите базовую валюту')).toBeInTheDocument();
    });

    it('App snapshot', () => {
        const app = render(
            <MemoryRouter>
                <App />
            </MemoryRouter>,
        );
        expect(app).toMatchSnapshot();
    });
});
