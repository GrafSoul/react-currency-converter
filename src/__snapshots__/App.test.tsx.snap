// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <header
          class="header"
        >
          <div
            class="headerContent"
          >
            <div
              class="headerName"
            >
              <a
                href="/"
              >
                <img
                  alt="Лого"
                  class="logo"
                  src="images/logo.png"
                />
              </a>
              <span
                class="name"
              >
                Конвертер валют
              </span>
            </div>
            <nav>
              <ul
                class="menu"
              >
                <li
                  class="menuItem"
                >
                  <a
                    class="active"
                    href="/"
                  >
                    Валюты
                  </a>
                </li>
                <li
                  class="menuItem"
                >
                  <a
                    class=""
                    href="/converter"
                  >
                    Конвертер
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </header>
        <main
          class="content"
        >
          <div
            class="currencies"
          >
            <form
              class="currencyForm"
            >
              <label
                class="labelCurrency"
                for="currency"
              >
                Выберите базовую валюту
              </label>
              <select
                class="selectCurrency"
                id="currency"
                name="currency"
              />
            </form>
            <div
              class="currenciesLoading"
            >
              <h3>
                Loading...
              </h3>
            </div>
            <div
              class="Toastify"
            />
          </div>
        </main>
        <footer
          class="footer"
        >
          <p
            class="copyRight"
          >
            © 
            2022
             | Конвертер валют | Created by
             
            <a
              class="copyLink"
              href="https://github.com/GrafSoul/react-currency-converter"
              id="footer-link"
              rel="noreferrer"
              target="_blank"
            >
              GrafSoul
            </a>
          </p>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <header
        class="header"
      >
        <div
          class="headerContent"
        >
          <div
            class="headerName"
          >
            <a
              href="/"
            >
              <img
                alt="Лого"
                class="logo"
                src="images/logo.png"
              />
            </a>
            <span
              class="name"
            >
              Конвертер валют
            </span>
          </div>
          <nav>
            <ul
              class="menu"
            >
              <li
                class="menuItem"
              >
                <a
                  class="active"
                  href="/"
                >
                  Валюты
                </a>
              </li>
              <li
                class="menuItem"
              >
                <a
                  class=""
                  href="/converter"
                >
                  Конвертер
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <main
        class="content"
      >
        <div
          class="currencies"
        >
          <form
            class="currencyForm"
          >
            <label
              class="labelCurrency"
              for="currency"
            >
              Выберите базовую валюту
            </label>
            <select
              class="selectCurrency"
              id="currency"
              name="currency"
            />
          </form>
          <div
            class="currenciesLoading"
          >
            <h3>
              Loading...
            </h3>
          </div>
          <div
            class="Toastify"
          />
        </div>
      </main>
      <footer
        class="footer"
      >
        <p
          class="copyRight"
        >
          © 
          2022
           | Конвертер валют | Created by
           
          <a
            class="copyLink"
            href="https://github.com/GrafSoul/react-currency-converter"
            id="footer-link"
            rel="noreferrer"
            target="_blank"
          >
            GrafSoul
          </a>
        </p>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
